neqo-http3/src/client_events.rs

```diff
@@ -88,19 +88,17 @@ impl RecvStreamEvents for Http3ClientEvents {
     }
 
     /// Add a new `Reset` event.
-    fn reset(&self, stream_id: u64, error: AppError, reset_type: ResetType) {
-        let local = match reset_type {
-            ResetType::App => return,
-            ResetType::Remote => false,
-            ResetType::Local => true,
+    fn recv_closed(&self, stream_id: u64, close_type: CloseType) {
+        if close_type == CloseType::Done {
+            return;
+        }
```

Comment:
If you move the remove_recv_side_events_for_stream_id (that name is too long) AFTER the match, then you don't need this if statement.

Reply:
That does not work because it needs to be called if type is ResetApp but not if it is Done. Both are exited in that match statement. It is not perfect. I will move the call to match statement.